flutter_todo/lib/realm/realm_services.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:flutter_todo/realm/schemas.dart';
+import 'package:realm/realm.dart';
+
+class RealmServices {
+  late Realm realm;
+  User? currentUser;
+  App app;
+
+  RealmServices(this.app) {
+    if (app.currentUser != null || currentUser != app.currentUser) {
+      currentUser ??= app.currentUser;
+      realm = initRealm(currentUser!);
+      updateSubscriptions();
+    }
+  }
+
+  Future<void> sessionOff(bool sessionOff) async {
+    if (sessionOff) {
+      realm.syncSession.pause();
+    } else {
+      realm.syncSession.resume();
+      await realm.syncSession.waitForDownload();
+      await realm.syncSession.waitForUpload();
+    }
+  }
+
+  Future<void> filterSwitch(bool isFilterOn) async {
+    updateSubscriptions(isFilterOn: isFilterOn);
+    await realm.subscriptions.waitForSynchronization();
+  }
+
+  Realm initRealm(User currentUser) {
+    Configuration config = Configuration.flexibleSync(currentUser, [Item.schema]);
+    Realm realm = Realm(
+      config,
+    );
+    return realm;
+  }
+
+  void updateSubscriptions({bool isFilterOn = false}) {
+    RealmResults<Item> query;
```

Comment:
should this have the `late` keyword before it?

Reply:
This is an old code, already.